meta-aku/Web_Test_VS
# Add testing library

In this PR

- Changed file structure to support TypeScript
- Added TypeScript and a testing library
- Some refactoring to code --> Changed structure a bit
- Fixed issues, where exported functions didn´t work when running the app
- Added ToDo comments, these are tasks to be done and some changes to be explained

How to test

1. Review the code changes --> There are quite many changes but try to look them through, to get familiar with code reviewing
2. Open project in VSCode
3. Open terminal in VSCode
4. On main branch run command `git pull --all` This will pull all new branches and all changes in them 
5. See the name of this branch on GitHub and copy it
6. In terminal run command `git checkout add-testing-library` (This will change your editor to use this branch instead of the main branch you have been working on)
7. Once you are on this branch (you can see branch name at the left bottom side in VSCode) in terminal run command `npm install` --> This will install the added libraries
(If you don´t have node and npm installed, you can install them from [here](https://nodejs.org/en/download/package-manager) )
8. Inside the `web_test-vs` folder, run command `npx tsc` --> This will compile the TypeScript files into js files that can be ran
9. Inside the `web_test-vs` folder, run command `npx http-server .` This will run a local web server to host the files
10. In browser type open `localhost:8080` --> You should see the files that are hosted
11. Open the public/home.html to access the application

Note
- When you run the npx commands for the first time, you might be asked to install the dependencies in order to use them
